tests/e2e/multicluster/multicluster_multiprimary_test.go

```diff
@@ -0,0 +1,479 @@
+							Should(ReturnNotFoundError(), "Istiod should not exist anymore")
+						common.CheckNamespaceEmpty(ctx, clPrimary, controlPlaneNamespace)
+						common.CheckNamespaceEmpty(ctx, clRemote, controlPlaneNamespace)
+						Success("Namespace is empty")
+					})
+				})
+			})
+		}
+	})
+
+	AfterAll(func(ctx SpecContext) {
+		// Delete the Sail Operator from both clusters
+		Expect(kubectl.DeleteNamespace(namespace, kubeconfig)).To(Succeed(), "Namespace failed to be deleted on Primary Cluster")
+		Expect(kubectl.DeleteNamespace(namespace, kubeconfig2)).To(Succeed(), "Namespace failed to be deleted on Remote Cluster")
+
+		// Delete the intermediate CA from both clusters
+		common.CheckNamespaceEmpty(ctx, clPrimary, namespace)
+		common.CheckNamespaceEmpty(ctx, clRemote, namespace)
+	})
+})
+
+// deploySampleApp deploys the sample app in the given cluster
+func deploySampleApp(ns string, istioVersion supportedversion.VersionInfo, kubeconfig string, kubeconfig2 string) {
+	// Create the namespace
+	Expect(kubectl.CreateNamespace(ns, kubeconfig)).To(Succeed(), "Namespace failed to be created")
+	Expect(kubectl.CreateNamespace(ns, kubeconfig2)).To(Succeed(), "Namespace failed to be created")
+
+	// Label the namespace
+	Expect(kubectl.Patch("", "namespace", ns, "merge", `{"metadata":{"labels":{"istio-injection":"enabled"}}}`)).
+		To(Succeed(), "Error patching sample namespace")
+	Expect(kubectl.Patch("", "namespace", ns, "merge", `{"metadata":{"labels":{"istio-injection":"enabled"}}}`, kubeconfig2)).
+		To(Succeed(), "Error patching sample namespace")
+
+	version := istioVersion.Version
+	// Deploy the sample app from upstream URL in both clusters
+	if istioVersion.Name == "latest" {
+		version = "master"
+	}
+	helloWorldURL := fmt.Sprintf("https://raw.githubusercontent.com/istio/istio/%s/samples/helloworld/helloworld.yaml", version)
+	Expect(kubectl.ApplyWithLabels(ns, helloWorldURL, "service=helloworld", kubeconfig)).To(Succeed(), "Sample service deploy failed on Primary Cluster")
+	Expect(kubectl.ApplyWithLabels(ns, helloWorldURL, "service=helloworld", kubeconfig2)).To(Succeed(), "Sample service deploy failed on Remote Cluster")
+
+	Expect(kubectl.ApplyWithLabels(ns, helloWorldURL, "version=v1", kubeconfig)).To(Succeed(), "Sample service deploy failed on Primary Cluster")
+	Expect(kubectl.ApplyWithLabels(ns, helloWorldURL, "version=v2", kubeconfig2)).To(Succeed(), "Sample service deploy failed on Remote Cluster")
+
+	sleepURL := fmt.Sprintf("https://raw.githubusercontent.com/istio/istio/%s/samples/sleep/sleep.yaml", version)
+	Expect(kubectl.Apply(ns, sleepURL, kubeconfig)).To(Succeed(), "Sample sleep deploy failed on Primary Cluster")
+	Expect(kubectl.Apply(ns, sleepURL, kubeconfig2)).To(Succeed(), "Sample sleep deploy failed on Remote Cluster")
+}
+
+// getPodName returns the pod name of the given deployment in the given namespace
+func getSleepPodName(ctx context.Context, cl client.Client, ns, deploymentName string) string {
```

Comment:
does `getSleepPodName` need to be its own function? Can just add the name of the pod as an additional argument to make it a little more generic. From the comment it looks like this used to be `getPodName`?

Reply:
Done, created the generic func `GetPodNameByLabel`